Source/Spatializer/CrossPlatform/SpatializerMixerPlugin.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Please note that this will only work on Unity 5.2 or higher.
+
+#include "AudioPluginUtil.h"
+#include "HrtfWrapper.h"
+#include "VectorMath.h"
+#include "mathutility.h"
+
+#include <cstring>
+
+namespace SpatialMixer
+{
+    struct EffectData
+    {
+        // History buffer used when the DSP buffer is smaller than the HRTF quantum
+        std::unique_ptr<float[]> HrtfHistoryBuffer;
+
+        // Current read offset into the history buffer
+        int ReadOffset;
+    };
+
+    int InternalRegisterEffectDefinition(UnityAudioEffectDefinition&)
+    {
+        return 0;
+    }
+
+    UNITY_AUDIODSP_RESULT UNITY_AUDIODSP_CALLBACK CreateCallback(UnityAudioEffectState* state)
+    {
+        auto effectdata = new EffectData;
+        std::memset(effectdata, 0, sizeof(EffectData));
+        state->effectdata = effectdata;
+
+        // If the DSP buffer size is smaller than the HRTF quantum, allocate a history buffer.
```

Comment:
I'm pretty sure you can excise all this. FLEX supports powers of two.

Reply:
Interesting, does this mean we don't need a history buffer at all?

Reply:
I think so? Depends how small Unity goes. As a data point, I was using 512 samples in the Wwise plugin with no problem.

Reply:
I'll test it out...

Reply:
Unity allows changing the dspbuffersize but HrtfEngine can't handle that re-init correctly right now. I investigated it a bit and I think there's a little bit of API rethinking required at the core engine API level (environment data along with the associated filters are currently static and don't reinit correctly). I'll file a bug/task against it but for now let's keep the history buffer.

Reply:
Interesting. Do we get a callback that the buffer size changed, or just a randomly new-sized buffer on the next frame?

Reply:
Let's quickly chat offline.